pkg/azuredisk/controllerserver.go

```diff
@@ -280,6 +280,21 @@ func (d *Driver) ControllerPublishVolume(ctx context.Context, req *csi.Controlle
 		return nil, status.Error(codes.InvalidArgument, "Volume capability not supported")
 	}
 
+	diskName, err := getDiskName(diskURI)
+	if err != nil {
+		return nil, status.Error(codes.NotFound, "Volume not found")
+	}
+
+	resourceGroup, err := getResourceGroupFromDiskURI(diskURI)
+	if err != nil {
+		return nil, status.Error(codes.NotFound, "Volume not found")
+	}
+
+	_, err = d.cloud.DisksClient.Get(ctx, resourceGroup, diskName)
+	if err != nil {
+		return nil, status.Error(codes.NotFound, "Volume not found")
```

Comment:
could you write like this, the original error message is important for diagnostics:
`return nil, status.Error(codes.NotFound, fmt.Sprintf("Volume not found, failed with error: %v", err))`